### ISSUE TITLE
Accessibility: Binderhub's Homepage

### ISSUE THREAD

**Issue by damli40** [jupyterhub/outreachy#65]
### What page is this for?

http://mybinder.org/

### WAVE accessibility report

**http://mybinder.org/ 's Home Page**

- 4 images do not have alternative text (problem). Add an alt attribute to the image. The attribute value should accurately and succinctly present the content and function of the image.(fix)

- The language of the document is not identified or a lang attribute value is invalid(problem). Identify the document language using the attribute with a valid value (e.g., ). Ensure that all lang attribute values are valid.(fix)

- Very low contrast between text and background colors in the page with some links. Adequate contrast of text is necessary for all users, especially users with low vision. This can be fixed by increasing the contrast between the foreground (text) color and the background color. Large text (larger than 18 point or 14 point bold) does not require as much contrast as smaller text, You can install a contrast checker on chrome to be sure it complies with WCAG contrast ratio(fix).

- No page regions or ARIA landmarks were found. The whole page was not built using semantic HTML and proper sectioning(problem). If the page has visual regions or significant page areas, ensure the regions are defined with header, nav, main, footer, etc. elements.(fix)

- The page does not have a first level heading(problem). The text "Turn a Git repo into a collection of interactive notebooks" should have a <h1></h1> tag as its the first heading on th page.

- On the "click to Donate" button, an event handler is present that may not be accessible.(problem) Ensure that critical functionality and content is accessible by using a device independent event handler (which responds to both keyboard and mouse) or by using both a mouse dependent and a keyboard dependent event handler.(fix)

- These are the results from the WAVE web app accessibility tester. All fixes can be done with just simple HTML and CSS.

![accessibility2](https://user-images.githubusercontent.com/57798516/196051145-09cc939b-8b66-479e-b3c7-0330ed5a6167.png)

### The HTML element to be changed

```html
<html>

<img id="logo" src="/static/logo.svg?v=fe52c40adc69454ba7536393f76ebd715e5fb75f5feafe16a27c47483eabf3311c14ed9fda905c49915d6dbf369ae68fb855a40dd05489a7b9542a9ee532e92b" width="390px">

<img id="badge" src="/static/images/badge_logo.svg?v=51b20aa16836ea83f5ed69194c660eb85c4c2c1e32565312baedb7d534e3ffcf592881dbbe3da441d8293ded842755c906b91fb4aadf15220cf48111ebf701c4">

<img class="icon" src="/static/images/markdown-icon.svg?v=d1115f582236340620dcc613052d5984f85f1742eed0e220e1a90c9f331bb71d3ef817bee9c6936ad6019c94cab66eb628528c929ca0bee3d25a4bdf634033cd">

<img class="icon clipboard" src="/static/images/copy-icon-black.svg?v=ff8d51629080acad6be94d3d35402a6318f452db7bcbac3d7617230ed70e7098a037b076ea7e5a662c599a9157563982e456a5f1854debe9fd2f1ef4067f3a3b" data-clipboard-target="#markdown-badge-snippet" alt="Copy markdown link to clipboard">

<a class="btn" style="width:fit-content;height:fit-content;padding:10px;background-color:#e66581;color:white;font-weight:bold;margin:-8px 0px;" onmouseover="this.style.backgroundColor='#d15b75'" onmouseout="this.style.backgroundColor='#e66581'" href="https://numfocus.salsalabs.org/donate-to-binder" target="_blank">
🤍 Donate to mybinder.org!
</a>

<h3>
Turn a Git repo into a collection of interactive notebooks
</h3>

<div style="display:flex;align-items:center;">
<div style="flex:1;text-align:center;">
Thanks to
Very low contrast
<a href="https://cloud.google.com/">
Google Cloud
</a>

,
Very low contrast
<a href="https://www.ovh.com/">
OVH
</a>

,
Very low contrast
<a href="https://notebooks.gesis.org">
GESIS Notebooks
</a>

and the
Very low contrast
<a href="https://turing.ac.uk">
Turing Institute
</a>

for supporting us! 🎉
</div>
//Device dependent event handler and Very low contrast

<a class="btn" style="width:fit-content;height:fit-content;padding:10px;background-color:#e66581;color:white;font-weight:bold;margin:-8px 0px;" onmouseover="this.style.backgroundColor='#d15b75'" onmouseout="this.style.backgroundColor='#e66581'" href="https://numfocus.salsalabs.org/donate-to-binder" target="_blank">
🤍 Donate to mybinder.org!
</a>
</div>

<button id="submit" class="btn-submit" type="submit">
launch
</button>

<span class="point point-orange">
1
</span>

<span class="point point-red">
2
</span>

<span class="point point-blue">
3
</span>
```

### Your proposed HTML (or CSS), after change

```html
// Language error fix 
<html lang="en">

//Alt text fix with images
<img id="logo" src="/static/logo.svg?v=fe52c40adc69454ba7536393f76ebd715e5fb75f5feafe16a27c47483eabf3311c14ed9fda905c49915d6dbf369ae68fb855a40dd05489a7b9542a9ee532e92b" width="390px" alt="Binderhub's Logo">

<img id="badge" src="/static/images/badge_logo.svg?v=51b20aa16836ea83f5ed69194c660eb85c4c2c1e32565312baedb7d534e3ffcf592881dbbe3da441d8293ded842755c906b91fb4aadf15220cf48111ebf701c4" alt='Image link to lanuch Binder'>

<img class="icon" src="/static/images/markdown-icon.svg?v=d1115f582236340620dcc613052d5984f85f1742eed0e220e1a90c9f331bb71d3ef817bee9c6936ad6019c94cab66eb628528c929ca0bee3d25a4bdf634033cd" alt="markdown icon">

<img class="icon clipboard" src="/static/images/copy-icon-black.svg?v=ff8d51629080acad6be94d3d35402a6318f452db7bcbac3d7617230ed70e7098a037b076ea7e5a662c599a9157563982e456a5f1854debe9fd2f1ef4067f3a3b" data-clipboard-target="#markdown-badge-snippet" alt="Copy markdown link to clipboard" alt ='Copy icon'>


//contrast error fix with the donate link and mouse handler fix with onfocus and onblur 

<a class="btn" style="width:fit-content;height:fit-content;padding:10px;background-color:#8C1730;color:white;font-weight:bold;margin:-8px 0px;" onmouseover="this.style.backgroundColor='#d15b75'" onmouseout="this.style.backgroundColor='#e66581'" href="https://numfocus.salsalabs.org/donate-to-binder" target="_blank"    onfocus="this.style.backgroundColor='#d15b75'" onblur="this.style.backgroundColor='#e66581'" >
🤍 Donate to mybinder.org!
</a>


//H1 error fix

<h1>
Turn a Git repo into a collection of interactive notebooks
</h1>

//more contrast fixes

<a href="https://cloud.google.com/" style="color: #1A3956">
Google Cloud
</a>

,
<a href="https://www.ovh.com/" style="color: #1A3956">
OVH
</a>

,
<a href="https://notebooks.gesis.org" style="color: #1A3956">
GESIS Notebooks
</a>

and the
<a href="https://turing.ac.uk" style="color: #1A3956">
Turing Institute
</a>

<button id="submit" class="btn-submit" type="submit" style="background-color:#7B4209" >
launch
</button>

<span class="point point-orange" style="color:#80450A">
1
</span>

<span class="point point-red" style="color:#982F49">
2
</span>

<span class="point point-blue" style="color:#275A7C">
3
</span>
```

### More info

The main changes in contrast were made using the WAVE tool inbuilt checker to make sure it aligns with WCAG contrast ratio.

The Page layout will need to be better by using semantic HTML elements such as: `<nav>, <header>, <main>, <aside>, <footer>,<section>` etc. This method organisez the page better and makes it easier to understand for everyone involved. This webpage was semingly also built with bootstrap which made inline styling necessary. 

The layout of the page was too packed, making it hard to read. This example below is a good one
![eg1](https://user-images.githubusercontent.com/57798516/196050941-c8eec181-16d2-4efa-bd26-782f4dca2a6f.png)

Having Events controlled by javascript in the html file is another subtle thing we can cut out to increase accessibilty, Trying to make all javascript and styling in their respective script and styling files.

**Comment by minrk** [comment]
Great! I'm curious about the js events. In principle, it makes sense to me, but can you find an example of one of the js events that would have this effect and perhaps a way to do it better?

I am growing familiar with accessibility for static a HTML page, but how to represent dynamic content accessibly is not something I've read about.

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
